visualizer/src/topology.js

```diff
@@ -196,18 +79,174 @@ module.exports = class Topology {
         this.initModule(d);
         return d;
     }
+}
+
+class jsPlumbConfig extends BaseConfig {
+    configureModules(modulemap, parent) {
+        for (const [moduleName, children] of Object.entries(modulemap)) {
+            let module = {}
+            module = this.newModule(moduleName);
+            if (children !== "") {
+                this.initParent(module);
+                this.configureModules(children, moduleName);
+            }
+            if (parent !== "") {
+                this.instance.addToGroup(`${parent}_group`, module);
+            }
+        }
+    }
+
+    createConnectionTypeMap() {
+        const connectionTypeMap = {};
+        for (const source of Object.keys(this.data.modules)) {
+            connectionTypeMap[source] = {};
+            for (const target of Object.keys(this.data.modules)) {
+                connectionTypeMap[source][target] = 5;
+            }
+        }
+        return connectionTypeMap;
+    }
+    
+    /**
+     * Utility method that return the nearest to 5 free number for a specific pair of source and target 
+     * and help to reduce lines overlaying.
+     * 
+     * This number is greater than 0 and less than 10. It's interpeted as the type of connection.
+     * 
+     * @private
+     * @this {jsPlumbConfig}
+     * @param {string} source - Source module.
+     * @param {string} target - Target module.
+     * @param {object} map - Map of connection type between each module.
+     * @return {number} - Connection Type.
+     */
+    getConnectionType(source, target, map) {
+        const tmp = map[source][target];
+        if (5 - tmp < 0) {
+            map[source][target] = 5 + (5 - tmp);

```

Comment:
What is the meaning of `(5 - tmp)` here?

Reply:
If we return a type that is greater than 5, in the next call we will return a type less than 5.
For example:
 ` for (let i = 0; i < 10; i++) { console.log(getConnectionType("A", "B", map)
}
`
prints `5 6 4 7 3 8 2 9 1 9`

Reply:
I cannot understand that code and what "5" or "10" or "7" means here. Even if you explain me now, everyone will forget it after a single week, and your code becomes non-maintainable.

Could you please use more expressive language features, like enumerations, named constants, etc.?